src/library/Uno.Toolkit.WinUI.Markup/ResourceValue.cs

```diff
@@ -0,0 +1,24 @@
+﻿using System;
+using Uno.Extensions.Markup;
+
+namespace Uno.Toolkit.Markup
+{
+	public readonly record struct ResourceValue<T>
+	{
+		public ResourceValue(string key, bool isThemeResource = false)
+		{
+			Key = key;
+			IsThemeResource = isThemeResource;
+		}
+
+		public string Key { get; }
+
+		public bool IsThemeResource { get; }
+
+		public static implicit operator Action<IDependencyPropertyBuilder<T>>(ResourceValue<T> resource) =>
+			resource.IsThemeResource ?
+				ThemeResource.Get<T>(resource.Key)
+				: StaticResource.Get<T>(resource.Key);
+
+	}
```

Comment:
rather than duplicating this across Themes and Toolkit let's just move this class to the core C# markup lib